homeassistant/components/device_tracker/swisscom.py

```diff
@@ -0,0 +1,101 @@
+"""
+Support for Swisscom routers (Internet-Box).
+
+For more details about this platform, please refer to the documentation at
+https://home-assistant.io/components/device_tracker.swisscom/
+"""
+import logging
+import threading
+from datetime import timedelta
+import requests
+
+from homeassistant.components.device_tracker import DOMAIN
+from homeassistant.const import CONF_HOST
+from homeassistant.util import Throttle
+
+# Return cached results if last scan was less then this time ago.
+MIN_TIME_BETWEEN_SCANS = timedelta(seconds=5)
+
+_LOGGER = logging.getLogger(__name__)
+
+
+def get_scanner(hass, config):
+    """Return the Swisscom device scanner."""
+    scanner = SwisscomDeviceScanner(config[DOMAIN])
+
+    return scanner if scanner.success_init else None
+
+
+class SwisscomDeviceScanner(object):
+    """This class queries a router running Swisscom Internet-Box firmware."""
+
+    def __init__(self, config):
+        """Initialize the scanner."""
+        # If the host is not configured, uses the default IP address.
+        self.host = config.get(CONF_HOST, '192.168.1.1')
```

Comment:
Can you please move this config format to a `PLATFORM_SCHEMA` and use voluptuous.